### DISCUSSION TITLE
Propagation/coercion for IsEnabled but not Visibility

### DISCUSSION THREAD

**Discussion by mattleibow** [dotnet/wpf#8774] [Q&A]
We recently got an issue and a PR in MAUI about making the Visibility property propagate: https://github.com/dotnet/maui/issues/19139#issuecomment-1927388990

I am just wondering if anyone knows why WPF decided that IsEnabled is propagated to all children, but not Visibility? Was there a specific decision or no-one ever asked for it?

**Comment by miloush** [comment]
Note that WPF has `Visibility` and `IsVisible` which are fundamentally different. The former represents the intent for that particular element, the latter is read-only, calculated visibility, which I guess is "propagated" in the sense you are thinking of (i.e. you cannot have `IsVisible` element until all parents are visible). Does that help?

**Reply by mattleibow** [reply]
Oh, let me check that IsVisible proeprty. Thanks!
